### PR TITLE
Add Snap fullnode daemon

### PR DESCRIPTION
**Pull request by mvines**

### CODE REVIEW

**Comment by mvines** [comment]
@garious / @rob-solana - I'm a little not proud about `multinode-demo/fullnode.sh` turning into a completely parallel way to launch a leader / validator.  I'd like to merge with the existing scripts either in this PR or in the not too distant future.

File: multinode-demo/node.sh [hunk 1]
```diff
@@ -0,0 +1,142 @@
+#!/bin/bash
+#
+# Runs a node
+#
+
+here="$(dirname "$0")"
```
Comment (rob-solana) [inline multinode-demo/node.sh]: double quotes around $() shouldn't be necessary

File: multinode-demo/node.sh [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                  Only required when mode=validator,validator-cuda,drone
+
+EOF
+  exit 1
+}
+
+
+MODE=
+if [[ -n "$1" ]]; then
+  MODE="$1"
```
Comment (rob-solana) [inline multinode-demo/node.sh]: double quotes inside [[ and in assignments unnecessary

File: multinode-demo/node.sh [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+validator-cuda|validator)
+  [[ -n "$LEADER_ADDRESS" ]] || usage "Error: leader_address not specified"
+
+  echo "Fetching configuration from $LEADER_ADDRESS:"
+  (
+    set -x
+    rm -rf "${DATA_DIR:?}/$LEADER_ADDRESS"
+    mkdir -p "$DATA_DIR/$LEADER_ADDRESS"
+    rsync -vrPz \
+      "rsync://$LEADER_ADDRESS"/solana/config/ \
```
[the remark this answers is not in the document]
Comment (mvines) [reply multinode-demo/node.sh]: sadly yes.  Otherwise rsync thinks `testnet.solana.com` is a local file 📉

**Comment by solana-grimes** [comment]
:scream: New commits were pushed while the automerge label was present.

**Pull request merged**